### PR TITLE
svg_loader: gradient inheritance implemented 

### PR DESCRIPTION
**Pull request by mgrudzinska**
Till now only the grad stops were inherited. Now all
grad attribs are.
Inheritance from different grad type is not supported.

@issue: https://github.com/thorvg/thorvg/issues/1209

### CODE REVIEW

**Comment by mgrudzinska** [comment]
split into two separate commits to make it easier to read

**Review by hermet — APPROVED** [review]
looks good to me. added some minor comments. @JSUYA's approval is welcomed.

File: src/loaders/svg/tvgSvgLoader.cpp [hunk 1]
```diff
@@ -2698,6 +2569,367 @@ static GradientFactoryMethod _findGradientFactory(const char* name)
...
+            }
+            //GradUnits set, coord not set directly
+            if (to->userSpace == from->userSpace) continue;
+            if (gradUnitSet && !coordSet) {
+                radialTags[i].tagInheritedRecalc(loader, to->radial, to->userSpace);
+            }
+        }
+    }
+
+    if (to->stops.count == 0) _cloneGradStops(to->stops, from->stops);
```
Comment (JSUYA) [inline src/loaders/svg/tvgSvgLoader.cpp:2637]: In the same context as now, can we add `stops` flag?
The reason for adding the flag is to check whether it is set to 0 or not set.
I think `to->stops.count == 0` is ambiguous.
Of course, it seem to be a no problem now. (Maybe?)

**Review by JSUYA — APPROVED** [review]
Declaration order changes caused a lot of diffs. It was a bit difficult to find real changes. hahaha...
LGTM :)

**Comment by mgrudzinska** [comment]
> Declaration order changes caused a lot of diffs. It was a bit difficult to find real changes. hahaha... LGTM :)

That's why I made two commits - one with changed order (without any logical changes) and one with the changes important to rev

**Pull request merged**